### PR TITLE
Feature/product autoselect

### PR DESCRIPTION
**Pull request by maciej-wakula-opuscapita**
Autoselect product if possible

If customer has only one product available - select it
otherwise try to preserve previously selected product
or clear product selection

### CODE REVIEW

File: src/client/components/Report/Report.jsx [hunk 1]
```diff
@@ -108,8 +108,23 @@ class Report extends Components.ContextComponent
         const config = response && response.config;
         const products = config ? config.products.map(product => ({ value: product.name, label: product.name })) : [];
 
+        var selectedProduct = null;
```
Comment (MisterFossey) [inline src/client/components/Report/Report.jsx:112]: Please use let and const instead of var

File: src/client/components/Report/Report.jsx [hunk 2]
```diff
@@ -108,8 +108,23 @@ class Report extends Components.ContextComponent
...
+
+        if(products){
+            if(products.length==1){
```
Comment (MisterFossey) [inline src/client/components/Report/Report.jsx:115]: 'products.length === 1' (Formatting)

File: src/client/components/Report/Report.jsx [hunk 3]
```diff
@@ -108,8 +108,23 @@ class Report extends Components.ContextComponent
...
+                selectedProduct = products[0].value;
+            }else{
+                var selectedProduct = this.state.selectedProduct;  // keep the same selection if possible
+                if(selectedProduct){
```
Comment (MisterFossey) [inline src/client/components/Report/Report.jsx:119]: if (..) - space after if

File: src/client/components/Report/Report.jsx [hunk 4]
```diff
@@ -108,8 +108,23 @@ class Report extends Components.ContextComponent
...
+                    const newSelection = products.filter(obj=> obj.value===selectedProduct);
```
Comment (MisterFossey) [inline src/client/components/Report/Report.jsx:120]: obj => obj.value === selectedProduct - Spaces for formatting

File: src/client/components/Report/Report.jsx [hunk 5]
```diff
@@ -108,10 +109,25 @@ class Report extends Components.ContextComponent
         const config = response && response.config;
         const products = config ? config.products.map(product => ({ value: product.name, label: product.name })) : [];
 
+        var selectedProduct = null;
+
+        if(products){
+            if(products.length==1){
+                selectedProduct = products[0].value;
+            }else{
```
Comment (MisterFossey) [inline src/client/components/Report/Report.jsx:117]: } else {

File: src/client/components/Report/Report.jsx (hunk 1 above)
Comment (maciej-wakula-opuscapita) [reply src/client/components/Report/Report.jsx]: var declares global scope and initialization works differently. Good to know. Thanks.
Resolved

File: src/client/components/Report/Report.jsx (hunk 2 above)
Comment (maciej-wakula-opuscapita) [reply src/client/components/Report/Report.jsx]: Fixed

File: src/client/components/Report/Report.jsx (hunk 4 above)
Comment (maciej-wakula-opuscapita) [reply src/client/components/Report/Report.jsx]: Fixed

File: src/client/components/Report/Report.jsx (hunk 5 above)
Comment (maciej-wakula-opuscapita) [reply src/client/components/Report/Report.jsx]: Fixed

**Pull request merged**